### PR TITLE
Added variadic filters, tests, and functions

### PR DESCRIPTION
**Pull request by hason**
Similar to #1159

```php
$env->addFunction(new Twig_SimpleFunction('foo', function ($a1, array $args = array()) {
}, array('is_variadic' => true));
```
```jinja
{{ foo(1, 2, a="a", b="b") }}
{# foo(1, array(0 => 2, "a" => "a", "b" => "b")); #}
```

### CODE REVIEW

**Comment by fabpot** [comment]
I would add only one argument as there is no difference between maps/sequences/hashes in PHP. Also, do we really need to add a configuration setting for that?

**Comment by hason** [comment]
@fabpot PR is updated

File: doc/advanced.rst [hunk 1]
```diff
@@ -224,6 +224,21 @@ through your filter::
...
+
+If you want to pass a variable number of positional or named arguments to the filter,
+set the ``is_variadic`` option to ``true``; Twig will pass the array of arbitrary arguments
+as the last argument to the filter call that is defined as an array with an empty default value::
+
+    $filter = new Twig_SimpleFilter('thumbnail', function ($file, array $options = array()) {
+        ...
+    }, array('is_variadic' => true));
+
+
```
Comment (fabpot) [inline doc/advanced.rst]: I suppose that this should also be documented for Functions and Tests.

You should also probably mention that the downside is that we cannot check anymore if the named arguments passed to the filter are valid or not as if they are not valid, they will end up in the option array.

**Comment by hason** [comment]
@fabpot The documentation is enhanced. Should I open a new PR for 1.x branch?

**Comment by hason** [comment]
Closing in favor of #1699

**Pull request closed without merging**